### PR TITLE
Init dataset

### PR DESCRIPTION
**Pull request by desmax74**
screen

### CODE REVIEW

File: roles/tpa_single_node/templates/manifests/init/dataset/Deployment.yaml.j2 [hunk 1]
```diff
@@ -0,0 +1,39 @@
+kind: Deployment
```
Comment (gildub) [inline roles/tpa_single_node/templates/manifests/init/dataset/Deployment.yaml.j2:2]: You need to add the following at the top :

```
apiVersion: apps/v1
```

**Review by gildub — COMMENTED** [review]
If this container is running only once for initialization of SBOMs and CEVs then it's  configuration doesn't need to go through "install_manifest.yml" to be resilient otherwise it should be deactivated when finished.

File: roles/tpa_single_node/templates/manifests/init/dataset/Deployment.yaml.j2 [hunk 2]
```diff
@@ -0,0 +1,40 @@
...
+          command: [ "/usr/bin/bash" ]
+          args:
+            - "-ce"
+            - |
+              set -e
+              /trust bombastic walker --sink http://sbom-pod --source file:/data/sbom
+              /trust vexination walker --sink http://vexination-api-pod:8080/api/v1/vex --source file:/data/csaf
+          env:
+            - name: OIDC_PROVIDER_CLIENT_ID
+              value: "{{ tpa_single_node_oidc_provider_client_id }}"
```
Comment (gildub) [inline roles/tpa_single_node/templates/manifests/init/dataset/Deployment.yaml.j2:31]:
```suggestion
            - name: OIDC_PROVIDER_CLIENT_ID
              value: "{{ tpa_single_node_oidc_provider_client_id }}"
```

File: roles/tpa_single_node/templates/manifests/init/dataset/Deployment.yaml.j2 [hunk 3]
```diff
@@ -0,0 +1,40 @@
...
+            - name: OIDC_PROVIDER_CLIENT_SECRET
+              valueFrom:
+                secretKeyRef:
+                  key: client_secret
+                  name: storage_secret
```
Comment (gildub) [inline roles/tpa_single_node/templates/manifests/init/dataset/Deployment.yaml.j2]:
```suggestion
            - name: OIDC_PROVIDER_CLIENT_ID
              valueFrom:
                secretKeyRef:
                  key: client_id
                  name: oidc_secret
            - name: OIDC_PROVIDER_CLIENT_SECRET
              valueFrom:
                secretKeyRef:
                  key: client_secret
                  name: oidc_secret               
                  
```

And you need to add the volumemounts and volumes accordingly with "oicd-secret".
Lookup the syntax in bombastic-api deployment manifest if needed.

File: roles/tpa_single_node/templates/manifests/init/dataset/Deployment.yaml.j2 [hunk 4]
```diff
@@ -0,0 +1,40 @@
...
+      containers:
+        - name: run
+          image: "{{ tpa_single_node_trustification_image }}"
+          imagePullPolicy: IfNotPresent
+          command: [ "/usr/bin/bash" ]
+          args:
+            - "-ce"
+            - |
+              set -e
+              /trust bombastic walker --sink http://sbom-pod --source file:/data/sbom
```
Comment (desmax74) [inline roles/tpa_single_node/templates/manifests/init/dataset/Deployment.yaml.j2]: This must be changed with the correct sbom url

**Review by gildub — APPROVED** [review]
LGTM

**Pull request merged**